prosopopee/prosopopee.py

```diff
@@ -229,129 +229,208 @@ def main():
-        os.path.realpath(os.path.join(os.getcwd(), "templates")),
-        os.path.join(os.path.split(os.path.realpath(__file__))[0], "themes", theme, "templates")
-    ]
+    if os.path.exists(os.path.join(os.getcwd(), "static")):
+        shutil.copytree(os.path.join(os.getcwd(), "static"), os.path.join(os.getcwd(), "build", gallery_path, "static"))
+    else:
+        shutil.copytree(
+            os.path.join(os.path.split(os.path.realpath(__file__))[0], "themes", theme, "static"),
+            os.path.join(os.getcwd(), "build", gallery_path, "static"))
+    return subgallery_templates
 
-    if theme != "exposure":
-        templates_dir.append(os.path.join(os.path.split(os.path.realpath(__file__))[0], "themes", "exposure", "templates"))
 
-    templates = Environment(loader=FileSystemLoader(templates_dir))
+def process_directory(gallery_name, settings, parent_templates, parent_gallery_path=False):
 
-    index_template = templates.get_template("index.html")
-    gallery_index_template = templates.get_template("gallery-index.html")
-    page_template = templates.get_template("page.html")
-    feed_template = templates.get_template("feed.xml")
+    if parent_gallery_path:
+        gallery_path = os.path.join(parent_gallery_path, gallery_name)
+    else:
+        gallery_path = gallery_name
 
-    # XXX recursively merge directories
-    if os.path.exists(os.path.join(os.getcwd(), "build", "static")):
-        shutil.rmtree(os.path.join(os.getcwd(), "build", "static"))
+    gallery_settings = yaml.safe_load(open(os.path.join(os.getcwd(), gallery_path, "settings.yaml"), "r"))
 
-    if os.path.exists(os.path.join(os.getcwd(), "static")):
-        shutil.copytree(os.path.join(os.getcwd(), "static"), os.path.join(os.getcwd(), "build", "static"))
+    error(isinstance(gallery_settings, dict), "Your %s should be a dict" % (os.path.join(gallery_name, "settings.yaml")))
+    error(gallery_settings.get("title"), "You should specify a title in %s" % (os.path.join(gallery_name, "settings.yaml")))
+
+    gallery_cover = {}
+
+    dirs = filter(lambda x: x not in (".", "..") and os.path.isdir(os.path.join(gallery_path, x)) and
+                  os.path.exists(os.path.join(os.getcwd(), gallery_path, x, "settings.yaml")),
+                  os.listdir(os.path.join(os.getcwd(), gallery_path)))
+
+    if not os.path.exists(os.path.join("build", gallery_path)):
+            os.makedirs(os.path.join("build", gallery_path))
+
+    if not gallery_settings.get("public", True):
+        build_gallery(settings, gallery_settings, gallery_path, parent_templates)
     else:
-        shutil.copytree(os.path.join(os.path.split(os.path.realpath(__file__))[0], "themes", theme, "static"), os.path.join(os.getcwd(), "build", "static"))
+        gallery_cover = create_cover(gallery_name, gallery_settings, gallery_path)
 
-    for gallery in dirs:
-        gallery_settings = yaml.safe_load(open(os.path.join(os.getcwd(), gallery, "settings.yaml"), "r"))
+        if dirs:
```

Comment:
An additional comment like "I have sub_galleries, let's build an index for them instead of a gallery" could be a good idea.